p2p/src/peer_manager/tests/ban.rs

```diff
@@ -417,3 +384,16 @@ async fn inbound_connection_invalid_magic_libp2p() {
 async fn inbound_connection_invalid_magic_mock() {
     inbound_connection_invalid_magic::<MockService>(make_mock_addr(), make_mock_addr()).await;
 }
+
+// TODO: Move to `p2p-test-utils`?
```

Comment:
It should probably go there and it should be different for mock and libp2p. That is because mock interface doesn't know what IPFS or meshsub are so while sharing the protocol names is convenient, it's also leaking implementation details of libp2p into mock.

Reply:
I have tried to do that, but I got the weird error:
```
error[E0308]: mismatched types
   --> p2p/src/peer_manager/tests/ban.rs:198:28
    |
198 |                 protocols: p2p_test_utils::default_protocols(),
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `net::types::protocol::Protocol`, found struct `p2p::net::types::protocol::Protocol`
    |
    = note: expected struct `std::collections::HashSet<net::types::protocol::Protocol>`
               found struct `std::collections::HashSet<p2p::net::types::protocol::Protocol>
```
I have only one idea: the compiler considers the types to be different because in one case it is reexported by the `p2p-test-utils` crate.

Reply:
But this function actually could be moved to `src/peer_manager/tests/mod.rs` so all `PeerManager` tests could use the function